test/Microsoft.IdentityModel.TestUtils/IdentityComparer.cs

```diff
@@ -1078,6 +1078,81 @@ public static bool AreTokenReadingResultsEqual(object object1, object object2, C
             return context.Merge(localContext);
         }
 
+        public static bool AreTokenDecryptingResultsEqual(object object1, object object2, CompareContext context)
```

Comment:
great to see the additional test coverage. 

is the performance roughly the same, improving or worse w/these changes?

Reply:
We will be getting to the benchmarks and optimisations shortly.